src/ray/core_worker/core_worker.cc

```diff
@@ -362,33 +406,33 @@ CoreWorker::CoreWorker(CoreWorkerOptions options, const WorkerID &worker_id)
     worker_context_.MaybeInitializeJobInfo(worker_context_.GetCurrentJobID(), job_config);
   }
 
-  local_raylet_client_ =
-      std::make_shared<raylet::RayletClient>(io_service_,
-                                             std::move(grpc_client),
-                                             options_.raylet_socket,
-                                             GetWorkerID(),
-                                             options_.worker_type,
-                                             worker_context_.GetCurrentJobID(),
-                                             options_.runtime_env_hash,
-                                             options_.language,
-                                             options_.node_ip_address,
-                                             &raylet_client_status,
-                                             &local_raylet_id,
-                                             &assigned_port,
-                                             options_.serialized_job_config,
-                                             options_.startup_token);
-
+  auto raylet_conn = std::make_unique<raylet::RayletConnection>(
+      io_service_, options_.raylet_socket, /*num_retries=*/-1, /*timeout=*/-1);
+  NodeID local_raylet_id;
+  int assigned_port = 0;
+  Status raylet_client_status = RegisterWorkerToRaylet(*raylet_conn,
```

Comment:
well, looks like RayletClient is not for sending RPC either, it's done here by CoreWorker. Then why do we have 2 classes RayletClient and RayletConnection? or, why do we want a *public* RayletConnection?

Reply:
I didn't change any accessibility for `RayletConnection` class.

Reply:
right it's not about visibility but intended API usage. Now, the caller (core worker) have to create a RayletConnection then use it to create a RayletClient, without any real usage of RayletConnection itself. Maybe we can cave a `static RayletClient Create()` func?

Reply:
> Now, the caller (core worker) have to create a RayletConnection then use it to create a RayletClient, without any real usage of RayletConnection itself.

I'm not sure what you mean, it's used in two places:
- `RegisterWorkerToRaylet` takes a raylet connection reference
- Then ownership passed to raylet client and used inside